ipv-api/src/main/java/uk/gov/di/authentication/ipv/helpers/IPVCallbackHelper.java

```diff
@@ -0,0 +1,341 @@
+    }
+
+    public AuthenticationSuccessResponse generateReturnCodeAuthenticationResponse(
+            AuthenticationRequest authRequest,
+            String clientSessionId,
+            UserProfile userProfile,
+            Session session,
+            ClientSession clientSession,
+            Subject rpPairwiseSubject,
+            String internalPairwiseSubjectId,
+            UserInfo userIdentityUserInfo,
+            String ipAddress,
+            String persistentSessionId)
+            throws UserNotFoundException {
+        LOG.warn("SPOT will not be invoked due to returnCode. Returning authCode to RP");
+        segmentedFunctionCall(
+                "saveIdentityClaims",
+                () -> saveIdentityClaimsToDynamo(rpPairwiseSubject, userIdentityUserInfo));
+        var authCode =
+                authorisationCodeService.generateAndSaveAuthorisationCode(
+                        clientSessionId, userProfile.getEmail(), clientSession);
+        var authenticationResponse =
+                new AuthenticationSuccessResponse(
+                        authRequest.getRedirectionURI(),
+                        authCode,
+                        null,
+                        null,
+                        authRequest.getState(),
+                        null,
+                        authRequest.getResponseMode());
+
+        var dimensions =
+                authCodeResponseService.getDimensions(
+                        session, clientSession, clientSessionId, false, false);
+
+        var subjectId = authCodeResponseService.getSubjectId(session);
+        auditService.submitAuditEvent(
+                IPVAuditableEvent.AUTH_CODE_ISSUED,
+                clientSessionId,
+                session.getSessionId(),
+                authRequest.getClientID().getValue(),
+                internalPairwiseSubjectId,
+                Objects.isNull(session.getEmailAddress())
+                        ? AuditService.UNKNOWN
+                        : session.getEmailAddress(),
+                ipAddress,
+                AuditService.UNKNOWN,
+                persistentSessionId,
+                pair("internalSubjectId", subjectId),
+                pair("isNewAccount", session.isNewAccount()),
+                pair("rpPairwiseId", rpPairwiseSubject.getValue()),
+                pair("nonce", authRequest.getNonce()),
+                pair("authCode", authCode));
+
+        var isTestJourney =
+                dynamoClientService.isTestJourney(clientSessionId, session.getEmailAddress());
+        LOG.info("Is journey a test journey: {}", isTestJourney);
+
+        cloudwatchMetricsService.incrementCounter("SignIn", dimensions);
+        cloudwatchMetricsService.incrementSignInByClient(
+                session.isNewAccount(),
+                clientSessionId,
+                clientSession.getClientName(),
+                isTestJourney);
+
+        authCodeResponseService.saveSession(false, sessionService, session);
```

Comment:
I wonder if we should session.setAuthenticated(true) before saving here. I think I would expect to be successfully auth'd even if my identity wasn't proven

Reply:
This happens inside the method